### DISCUSSION TITLE
how to set a combobox model dynamically from rust ?

### DISCUSSION THREAD

**Discussion by ffred** [slint-ui/slint#5109] [Q&A]
I have a `Vec<slint::SharedString>` that I want to send to a in property to populate a combobox, but that's not the good type.
```
mismatched types
expected struct `ModelRc<slint::SharedString>`
   found struct `Vec<slint::SharedString>`
```
I'm still learning Rust and Slint, so it seems complicated for me right now.
I can't set an Array, because the lenght of data I want in the Combobox is define at runtime.
didn't find a Combobox example witch could show me one way to do that...

**Comment by bjorn** [comment]
Given your `Vec<slint::SharedString>`, you can turn it into a `ModelRc<slint::SharedString>` using:
```rust
Rc::new(VecModel::from(string_vec)).into()
```
Some other approaches for constructing the `ModelRc` are documented here: https://releases.slint.dev/1.5.1/docs/rust/slint/struct.modelrc

**Comment by ffred** [comment]
thank you !!
